### ISSUE TITLE
Custom command not called

### ISSUE THREAD

**Issue by Daenara** [rhasspy/rhasspy#50]
I posted this a while ago in the pre release forum topic but I thought actually creating an issue might be better. After migrating to 2.5 (both pre and release) my custom command script is not called anymore. I have it set up the same way as I did in 2.4 in my profile folder, the script is executable and I can run it from within the docker (after manually installing pip). I do not get any error in the rhasspy log but I can tell that my script does not run because I have my own log that should get a new entry in my log as soon as the script is called (directly after the imports).

Here are my settings:
![image](https://user-images.githubusercontent.com/3619315/83946228-bb629f80-a80f-11ea-96ac-161632b98741.png)
I also tried without the curly brackets around the profile dir.

Here is what the log says when I recognize the intent with handle checked:
![image](https://user-images.githubusercontent.com/3619315/83946270-01b7fe80-a810-11ea-85e8-474e2d7e8fd4.png)

I also did tests in "productive" mode, meaning using a wakeword and then asking the question in case the checkbox is broken but the result was the same.

I do know that I can use mqtt directly but I try to be as modular as possible so I want to support every option available and also, since i just came from 2.4 I do not have a mqtt script written yet.

Daenara

**Comment by synesthesiam** [comment]
Thanks, I'll take a look. I haven't tested the local command much, and I may have broken something in the last few months.

**Comment by DeadEnded** [comment]
It appears I am having the same issue.
Followed the docs, using the example handler.py and it still doesn't work.

Happy to do any testing I can - but I'm pretty inexperienced, so might need a bit of detail 😄 .

DeadEnd

**Comment by DeadEnded** [comment]
I was digging in the code trying to understand more, and I think I found something but don't know if it would cause a problem...

In the intent handler, there is a reference to satellites... but at least in the GUI there is no way to set them... I wasn't sure if this would cause it to error when trying to use a local intent handler.

rhasspy-supervisor/rhasspysupervisor/hasspysupervisor/__init__.py
Lines 140-145
```
    # Intent Handling
    handle_system = profile.get("handle.system", "dummy")
    if handle_system not in {"dummy", "hermes"}:
        satellite_site_ids = str(profile.get("handle.satellite_site_ids", "")).split(
            ","
        )
```
Since I'm pretty dev dumb... would this cause an error if the value doesn't exist and doesn't have a fallback default?

**Comment by DeadEnded** [comment]
Something more interesting... I watched the docker logs... and was getting this error:
```
[ERROR:2020-08-04 17:02:10,967] rhasspyremote_http_hermes: handle_intent
Traceback (most recent call last):
  File "/usr/lib/rhasspy/rhasspy-remote-http-hermes/rhasspyremote_http_hermes/__init__.py", line 798, in handle_intent
    stderr=asyncio.subprocess.PIPE,
  File "/usr/lib/python3.7/asyncio/subprocess.py", line 217, in create_subprocess_exec
    stderr=stderr, **kwds)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 1529, in subprocess_exec
    bufsize, **kwargs)
  File "/usr/lib/python3.7/asyncio/unix_events.py", line 190, in _make_subprocess_transport
    **kwargs)
  File "/usr/lib/python3.7/asyncio/base_subprocess.py", line 37, in __init__
    stderr=stderr, bufsize=bufsize, **kwargs)
  File "/usr/lib/python3.7/asyncio/unix_events.py", line 775, in _start
    universal_newlines=False, bufsize=bufsize, **kwargs)
  File "/usr/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.7/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '${RHASSPY_PROFILE_DIR}/handler.py': '${RHASSPY_PROFILE_DIR}/handler.py'
```
So I didn't use the reference, and instead typed out `/profiles/en/handler.py`
and now IT WORKS!
So it appears there is an issue with using the variable paths with the intent handler.

@Daenara can you replace your `${RHASSPY_PROFILE_DIR}` with an absolutely path (/profiles/en in the container).  See that fixes it for you as well!

DeadEnd

**Comment by Daenara** [comment]
That was it, it works now. I never looked at the full log, just the part that it showed in the webgui so I never caught that. Thanks for figuring that out @DeadEnded

**Comment by DeadEnded** [comment]
So I have to say... I am VERY excited... I think I found the problem and have created a pull request for the solution.
After a LOT of digging to understand where everything is (total noob) I was able to compare the intent handler with the wav file directory fields.  I found the line of code that expands the environmental variables - and it wasn't present for the intent handler.

I added it in my container and after a restart it appears to be working.
This was just a 1 line change, but I am very happy to have been able to troubleshoot it and find the solution.
I use a lot of open source software, so I am always trying to contribute.

Here is the pull request - not sure what the requirements are... I created it against Master which I expect is not the "correct" way... so I might have to re-create it for one of the tags... unless the maintainers allow it because it is a single line addition - so very little to review.
https://github.com/rhasspy/rhasspy-supervisor/pull/4

Cheers!
DeadEnd
